File: Ruby/find_the_unique_number.rb
def find_uniq(arr)
  key = {}
  res = 0
  arr.each { |n| key[n] = 0 }
  arr.each { |n| key[n] += 1 }
  arr.each { |n| res = n if key[n] == 1 }
  return res
end

p find_uniq([1, 1, 1, 2, 1, 1])
p find_uniq([0, 0, 0.55, 0, 0])

File: Python/find_the_unique_number.py
def find_uniq(arr):
    key = {}
    res = 0
    for i in arr:
        key[i] = 0
    for j in arr:
        key[j] += 1
    for k in arr:
        if key[k] == 1:
            res = k
    return res


find_uniq([1, 1, 1, 2, 1, 1])
find_uniq([0, 0, 0.55, 0, 0])
